src/DateTimeFormatter.php

```diff
@@ -4,99 +4,68 @@
 
 namespace Laminas\Filter;
 
-use DateTime;
+use DateTimeImmutable;
 use DateTimeInterface;
+use DateTimeZone;
+use Exception;
+use Laminas\Filter\Exception\InvalidArgumentException;
 use Throwable;
-use Traversable;
 
+use function date_default_timezone_get;
 use function is_int;
-use function is_string;
 
 /**
  * @psalm-type Options = array{
- *     format?: string,
- *     ...
+ *      format?: non-empty-string,
+ *      timezone?: non-empty-string,
  * }
- * @extends AbstractFilter<Options>
+ * @implements FilterInterface<string>
  */
-final class DateTimeFormatter extends AbstractFilter
+final class DateTimeFormatter implements FilterInterface
 {
     /**
      * A valid format string accepted by date()
-     *
-     * @var string
      */
-    protected $format = DateTime::ISO8601;
+    private readonly string $format;
 
     /**
-     * Sets filter options
-     *
-     * @param array|Traversable $options
+     * A valid timezone string
      */
-    public function __construct($options = null)
-    {
-        if ($options !== null) {
-            $this->setOptions($options);
-        }
-    }
+    private readonly DateTimeZone $timezone;
 
     /**
-     * Set the format string accepted by date() to use when formatting a string
-     *
-     * @param  string $format
-     * @return self
+     * @param Options $options
+     * @throws Exception
      */
-    public function setFormat($format)
+    public function __construct(array $options = [])
     {
-        $this->format = $format;
-
-        return $this;
+        $this->format   = $options['format'] ?? DateTimeInterface::ATOM;
+        $this->timezone = new DateTimeZone($options['timezone'] ?? date_default_timezone_get());
     }
 
     /**
      * Filter a datetime string by normalizing it to the filters specified format
      *
-     * @param  DateTime|string|int|mixed $value
-     * @throws Exception\InvalidArgumentException
+     * @inheritDoc
      */
     public function filter(mixed $value): mixed
     {
         try {
-            $result = $this->normalizeDateTime($value);
+            if (! $value instanceof DateTimeInterface) {
+                $value = is_int($value) ? '@' . (string) $value : $value;
+                $value = new DateTimeImmutable($value, $this->timezone);
+            } else {
+                $value = new DateTimeImmutable($value->format($this->format), $this->timezone);
+            }
         } catch (Throwable $e) {
-            // DateTime threw an exception, an invalid date string was provided
-            throw new Exception\InvalidArgumentException('Invalid date string provided', $e->getCode(), $e);
+            throw new InvalidArgumentException('Invalid date string provided', $e->getCode(), $e);
```

Comment:
`$e->getCode()` should be cast to `(int)` - the baseline will need refreshing too